sdk/identity/azure-identity/src/azure_cli_credential.cpp

```diff
@@ -0,0 +1,654 @@
+
+  // Typically token json is just a bit less than 2KiB.
+  // The best buffer size is the one that lets us to read it in one go.
+  // (Should it be smaller, we will succeed as well, it'll just take more iterations).
+  std::vector<std::string::value_type> processOutputBuf(2 * 1024);
+
+  auto willHaveMoreData = true;
+  do
+  {
+    // Check if we should terminate
+    {
+      if (context.IsCancelled())
+      {
+        shellProcess.Terminate();
+        throw std::runtime_error("Context was cancelled before Azure CLI process was done.");
+      }
+
+      if (std::chrono::steady_clock::now() > terminateAfter)
+      {
+        shellProcess.Terminate();
+        throw std::runtime_error("Azure CLI process took too long to complete.");
+      }
+    }
+
+    std::string::size_type bytesRead = 0;
+    if (pipe.NonBlockingRead(processOutputBuf, bytesRead, willHaveMoreData))
+    {
+      output.insert(output.size(), processOutputBuf.data(), bytesRead);
+    }
+    else if (willHaveMoreData)
+    {
+      std::this_thread::sleep_for(std::chrono::milliseconds(200)); // Value has no special meaning.
+    }
+  } while (willHaveMoreData);
+
+  return output;
+}
+
+#if defined(AZ_PLATFORM_WINDOWS)
+void ThrowIfApiCallFails(BOOL apiResult, std::string const& errMsg)
+{
+  // LCOV_EXCL_START
+  if (!apiResult)
+  {
+    throw std::runtime_error(
+        errMsg + ": " + std::to_string(GetLastError())
+
+    );
+  }
+  // LCOV_EXCL_STOP
+}
+#else
+void ThrowIfApiCallFails(int apiResult, std::string const& errMsg)
+{
+  // LCOV_EXCL_START
+  if (apiResult != 0)
+  {
+    throw std::runtime_error(
+        errMsg + ": " + std::to_string(apiResult) + " (errno: " + std::to_string(errno) + ")");
+  }
+  // LCOV_EXCL_STOP
+}
+#endif
+
+OutputPipe::OutputPipe()
+{
+#if defined(AZ_PLATFORM_WINDOWS)
+  SECURITY_ATTRIBUTES pipeSecurity = {};
+  ZeroMemory(&pipeSecurity, sizeof(decltype(pipeSecurity)));
+  pipeSecurity.nLength = sizeof(decltype(pipeSecurity));
+  pipeSecurity.bInheritHandle = TRUE;
+  pipeSecurity.lpSecurityDescriptor = NULL;
+
+  ThrowIfApiCallFails(
+      CreatePipe(&m_readHandle, &m_writeHandle, &pipeSecurity, 0), "Cannot create output pipe");
+
+  ThrowIfApiCallFails(
+      SetHandleInformation(m_readHandle, HANDLE_FLAG_INHERIT, 0),
+      "Cannot ensure the read handle for the output pipe is not inherited");
+
+  // We use OVERLAPPED when we ReadFile(), so that the call is non-blocking, which lets us to also
+  // poll whether we should terminate the process.
+  ZeroMemory(&m_overlapped, sizeof(decltype(m_overlapped)));
+#else
+  m_fd.push_back(-1);
+  m_fd.push_back(-1);
+
+  ThrowIfApiCallFails(pipe(m_fd.data()), "Cannot create output pipe");
+  ThrowIfApiCallFails(
+      fcntl(m_fd[0], F_SETFL, O_NONBLOCK), "Cannot set up output pipe to have non-blocking read");
+
+#endif
+}
+
+OutputPipe::~OutputPipe()
+{
+#if defined(AZ_PLATFORM_WINDOWS)
+  if (m_writeHandle != NULL)
+  {
+    static_cast<void>(CloseHandle(m_writeHandle));
```

Comment:
Does static analysis fail if you don't have the `static_cast<void>` here?

Reply:
I am not sure about this specific case. I definitely was getting warnings in some other places when functions were returning values. So I wrap them all. And it is clear that there never was an intention to check the result.